core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -570,28 +576,53 @@ private[spark] class BlockManager(
     preferredLocs ++ otherLocs
   }
 
-  private def doGetRemote(blockId: BlockId, asBlockResult: Boolean): Option[Any] = {
+  /**
+   * VisibleForTesting
+   */
+  private[spark] def doGetRemote(
+      blockId: BlockId,
+      blockTransferService: BlockTransferService,
```

Comment:
this interface is really clunky. I would refactor your tests a little to pass in your `MockBlockTransferService` when you `makeBlockManager`. Then (1) you don't need to expose this method and (2) you don't need to pass in a transfer service here, which is strange.